firebase-common/api.txt

```diff
@@ -59,17 +59,24 @@ package com.google.firebase {
   }
 
   public final class Timestamp implements java.lang.Comparable<com.google.firebase.Timestamp> android.os.Parcelable {
-    ctor public Timestamp(long, int);
-    ctor protected Timestamp(@NonNull android.os.Parcel);
-    ctor public Timestamp(@NonNull java.util.Date);
-    method public int compareTo(@NonNull com.google.firebase.Timestamp);
+    ctor public Timestamp(long seconds, int nanoseconds);
+    ctor public Timestamp(@NonNull android.os.Parcel source);
```

Comment:
This constructor's visibility has been upgraded from `protected` to `public`. Was this intentional?

Reply:
I was thinking about this, and actually I think it should be removed. It was `protected` before, but `Timestamp` was a `final` class- so it was technically `private`. As such, there's no need to pollute the constructors with it. It also had the side effect of skipping out on the error checking, so removing it and inlining it at the call site should fix this.

Reply:
Makes sense. I'm fine with deleting the constructor, especially since this change is happening as part of a major version bump.

Reply:
I'm un-resolving this comment as a reminder to remove the constructor.

Reply:
The most common way, even used as the example in the documentation of `Parcelable`, is to have a private/protected constructor that takes `Parcel` and make CREATOR call it. That's the pattern seen here. Additionally, given that the system will only call CREATOR using a parcel created by a valid object, it's safe to bypass validation.

While an implementation detail, I'd rather keep that code pattern out of consistency.

Reply:
While technically true, are we sure that's the intent there? While it hasn't been updated for about a year, it seems parcelize was going in a non boiler-plate direction with the [kotlin-parcelize plugin](https://developer.android.com/kotlin/parcelize). While the example uses a private primary constructor, it might have just been what came to mind at the time. Especially since it was initially designed with Java in mind.

Reply:
It's indeed a more java-ish pattern, as it's the fact that this class has a bunch of constructors. 

In any case, it doesn't make much difference which way we choose to go since it's effectively an implementation detail, so feel free to ignore this comment :-). I wanted to add some extra context on why there's that constructor. I remember having seeing the pattern in other old codebases.